pages/index.vue

```diff
@@ -19,54 +19,50 @@ export default {
   components: {CardList},
   data() {
     return {
-      isLastPage: true,
+      characters: [],
       totalCharacters: 0,
+      isLastPage: true,
       loading: true,
-      characters: [],
       filters: {
+        nameStartsWith: undefined,
         offset: 0,
         limit: 20,
         orderBy: "name",
       },
     };
   },
-
-  mounted() {
-    this.fetchData();
-  },
-
-  methods: {
-    fetchData() {
-      this.loading = true;
-      Character.list(this.filters).then((response) => {
-        if(response.code === 200) {
-          this.characters = response?.data?.results
-          this.totalCharacters = response?.data?.total
-        } else this.characters = []  
+  
+  async fetch() {
+      if(this.filters.offset === 0) { this.loading = true }
+      await(Character.list(this.filters)
+      .then((response) => {
+        const data = response.code === 200 ? response?.data?.results : [];
+        if(this.filters.offset === 0) { this.characters = [] }
+        this.characters = this.characters.concat(data);
+        this.totalCharacters = response?.data?.total  
       }).finally(() => {
           this.loading = false
           this.isLastPage = this.filters.offset + this.filters.limit >= this.totalCharacters
-        });
+        }))
     },
+    
+  watch: {
+    filters: {
+      deep: true,
+      handler() {
+        this.$fetch()
+      }
+    }
+  },
 
+  methods: {
     appendNextPage() {
       this.filters.offset += 20;
-      Character.list(this.filters).then((response) => {
-        const data = response.code === 200 ? response?.data?.results : [];
-        this.characters = this.characters.concat(data);
-        this.isLastPage = this.filters.offset + this.filters.limit >= this.totalCharacters
-      });
     },
+
     debounceSearch: _.debounce(function (e) {
       this.filters.offset = 0;
-      this.loading = true
-        if (e !== "") {
-          this.filters.nameStartsWith = e;
-          this.fetchData();
-        } else {
-          delete this.filters["nameStartsWith"];
-          this.fetchData();
-        }
+      this.filters.nameStartsWith = e || undefined
```

Comment:
Não seria só uma barra?

```suggestion
      this.filters.nameStartsWith = e | undefined
```